### PR TITLE
Feature keys and doors

### LINKED ISSUE #40: Doors
This task consists of the implementation of doors in our game, as well as the interaction of keys with doors (Door opens when player has key/acquires key)

### LINKED ISSUE #100: Add  Keys to game

### PR DESCRIPTION
**Pull request by WytzeElhorst**
Door.java and Key.java, both with tests + models

This closes #40 and closes #100

### CODE REVIEW

**Comment by Geweldig** [comment]
Hello.png should not be uploaded to github.

File: src/main/assets/Materials/newMaterial.j3m [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Material My Material : Common/MatDefs/Misc/Unshaded.j3md {
+     MaterialParameters {
+        ColorMap : Textures/key.png
+     }
+    AdditionalRenderState {
+    }
+}
```
Comment (ChielBruin) [inline src/main/assets/Materials/newMaterial.j3m]: keyMaterial.j3m is probably a beter name.

File: src/main/java/nl/tudelft/contextproject/Door.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.Node;
+import com.jme3.scene.Spatial;
+import com.jme3.scene.shape.Box;
+
+/**
+ * Class representing a door.
+ */
+public class Door implements Drawable {
```
Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/Door.java]: Door should extend Entity as it is an entity in the game.

**Comment by ChielBruin** [comment]
Hello.png is still added. I also recommend to change the filename of newMaterial. The rest looks good!

**Comment by Geweldig** [comment]
I agree completely with Chiel.

**Comment by Geweldig** [comment]
Can be merged in my opinion!

**Comment by ChielBruin** [comment]
The commit says renamed, but the diff shows removed?

**Comment by WytzeElhorst** [comment]
last commit instead of renaming the file i created a copy with the correct name, so now the file with the wrong name got deleted

**Comment by ChielBruin** [comment]
Then I will merge the pull request

**Pull request merged**